File: backend/app/data/chassis_colors.py
"""Chassis color presets — must stay aligned with frontend/lib/chassisColors.js and CMS seed."""

from __future__ import annotations

CHASSIS_COLOR_IDS = frozenset({
    "candy-red-white",
    "candy-blue-white",
    "candy-purple-white",
    "custom",
    # Disruptor Paramotor
    "white-red-candy",
    "white-purple-candy",
    "white-blue-candy",
    "customized",
    # Legacy slugs (older frontend / Disruptor seed builds)
    "red-candy-white",
    "blue-candy-white",
    "purple-candy-white",
    "white",
    "red-candy",
    "blue-candy",
    "purple-candy",
    "candy-red",
    "candy-blue",
    "candy-purple",
    "grey",
})

CUSTOM_COLOR_ID = "custom"
CUSTOM_COLOR_SURCHARGE = 100.0


def chassis_color_price(color_id: str | None, colors_catalog: dict | None = None) -> float:
    if not color_id:
        return 0.0
    if color_id in (CUSTOM_COLOR_ID, "customized"):
        if colors_catalog and color_id in colors_catalog:
            return float(colors_catalog[color_id] or CUSTOM_COLOR_SURCHARGE)
        return CUSTOM_COLOR_SURCHARGE
    if color_id in CHASSIS_COLOR_IDS:
        return 0.0
    if colors_catalog and color_id in colors_catalog:
        return float(colors_catalog[color_id] or 0.0)
    return 0.0


def is_allowed_chassis_color(color_id: str | None, colors_catalog: dict | None = None) -> bool:
    if not color_id:
        return True
    if color_id == CUSTOM_COLOR_ID:
        return True
    if color_id in CHASSIS_COLOR_IDS:
        return True
    if colors_catalog and color_id in colors_catalog:
        return True
    return False

File: frontend/lib/chassisColors.js
/** Shared chassis color presets — first step on all trike configurators. */

export const CUSTOM_COLOR_ID = 'custom'
export const CUSTOM_COLOR_SURCHARGE = 100

/** IDs must match CMS seed (candy-*-white) and backend chassis_colors.py */
export const CHASSIS_COLOR_PRESETS = [
  { id: 'candy-red-white', name: 'Red Candy & White', hex: '#e74c3c', accent: '#ffffff' },
  { id: 'candy-blue-white', name: 'Blue Candy & White', hex: '#3498db', accent: '#ffffff' },
  { id: 'candy-purple-white', name: 'Purple Candy & White', hex: '#9b59b6', accent: '#ffffff' },
]

export function resolveChassisColorLabel(colorId, customText = '') {
  if (colorId === CUSTOM_COLOR_ID) {
    const trimmed = String(customText || '').trim()
    return trimmed ? `Custom: ${trimmed}` : 'Custom color'
  }
  return CHASSIS_COLOR_PRESETS.find((c) => c.id === colorId)?.name || colorId || ''
}

export function chassisColorSurcharge(colorId) {
  return colorId === CUSTOM_COLOR_ID ? CUSTOM_COLOR_SURCHARGE : 0
}
